client/src/components/chat-bubble/chat-bubble.js

```diff
@@ -0,0 +1,33 @@
+import Component from '@/components/Component';
+
+import styles from '@/styles/components/chat-bubble/chat-bubble.module.scss';
+
+function stringPreprocess(string) {
+  return string.replace(/ /g, '&nbsp;').replace(/\r\n/g, '\n');
+}
```

Comment:
```suggestion
function stringPreprocess(str) {
  return str.trim().replace(/ /g, '&nbsp;').replace(/\r?\n/gm, '<br />');
}
```

음.. 매개변수로 `string`을 써도 되기는 하는데 뭔가 type 같아서 꺼림직합니다.(ts는 예약어로 먹던가... 아닌가..)
`str`로 변경했으면 좋겠습니다.

`str.trim()`을 통해서 앞 뒤로 불필요한 공백을 제거하는게 어떨까요?

`replace(/\r?\n/gm, '<br />')` \r\n을 잡아 주실때는 \r이 없는 경우도 있기 때문에 `/\r?\n/`으로 잡아주셔야 합니다.
또한 new line의 경우는 정규표현식의 기본동작이 여러줄에서 한줄씩 잡기 때문에 /m (multiline) flat를 활성화 해주셔야 합니다.